src/app/calculator/calculator.component.ts

```diff
@@ -320,7 +320,7 @@ export class CalculatorComponent implements OnInit {
         if (e.data == null)
             return;
 
-        let validRegEx = allowDecimalPoint ? /^[\d.]$/ : /^[\d]$/;
+        let validRegEx = allowDecimalPoint ? /^\d|.+$/ : /^\d+$/;
```

Comment:
/^\d|.+$/ allows either 1 digit character or multiple of ANY character, due to the dot being a token in Regular Expressions that translates to ANY character in this context, the brackets made it a list of valid characters, and therefore the dot gets interpreted as the character "." instead of "any".
/^[\d.]+$/ would be the correct way of restricting it to the Character "." or 0-9 multiple times.
Change that and I'll approve this as soon as I'm back

Reply:
Cheers didn't pick that one up.